src/sidebar/components/annotation-action-bar.js

```diff
@@ -0,0 +1,109 @@
+'use strict';
+
+const propTypes = require('prop-types');
+const { createElement } = require('preact');
+
+const { withServices } = require('../util/service-context');
+const { isShareable, shareURI } = require('../util/annotation-sharing');
+
+const AnnotationActionButton = require('./annotation-action-button');
+const AnnotationShareControl = require('./annotation-share-control');
+
+/**
+ * A collection of `AnnotationActionButton`s in the footer area of an annotation.
+ */
+function AnnotationActionBar({
+  annotation,
+  isPrivate,
+  onDelete,
+  onEdit,
+  onFlag,
+  onReply,
+  groups,
+  permissions,
+  session,
+  settings,
+}) {
+  // Is the current user allowed to take the given `action` on this annotation?
+  const userIsAuthorizedTo = action => {
+    return permissions.permits(
+      annotation.permissions,
+      action,
+      session.state.userid
+    );
+  };
+
+  const showDeleteAction = userIsAuthorizedTo('delete');
+  const showEditAction = userIsAuthorizedTo('update');
+  // Anyone may flag an annotation except the annotation's author.
```

Comment:
This is not quite true. Anonymous users can see a "flag" action but get presented with a message telling them to log in if they try to use it.